pkg/storage/cacher_test.go

```diff
@@ -302,6 +296,39 @@ func TestWatch(t *testing.T) {
 		}
 	}
 
+	// Now test watch from "now".
+	nowWatcher, err := cacher.Watch("pods/ns/foo", 0, storage.Everything)
```

Comment:
I think we still need a test to test the too-old-error scenario when resourceVersion != 0

Is that covered elsewhere, it looked like previously we were using 0 as that test which is no longer valid.